### ISSUE TITLE
Stabilise learning to predict relations

### ISSUE THREAD

**Issue by jmsfltchr** [graknlabs/kglib#100]
## Problem to Solve

During relation prediction we see that the focus of the algorithm is to correctly classify **all** nodes and edges. Of these there are typically far more pre-existing nodes and edges that we don't care about than that we do.

## Current Workaround

None

## Proposed Solution

Possible options:
- Update the loss function of the learner
- Propagate preexistence through the neural network to reduce the burden on the network
- Filter out Unnecessary nodes in the decoder stage of the network, prior to calculating loss
- Reduce the number of output classes from `[preexists, does not exist, does exist]` to `[does not exist, does exist]`

**Comment by jmsfltchr** [comment]
Partially solved by #101 , but the number of output classes is still 3, although the loss function now ignores `preexists`
